Gruntfile.js

```diff
@@ -696,4 +675,51 @@ module.exports = function(grunt) {
 		}
 		return src;
 	}
+
+	/**
+	 * Transform JS inside some file into a `case` inside a `switch` statement 
+	 * whilst at the same time compiling the list of supported language codes.
+	 * @param {Set<string>} codes
+	 * @returns {Function}
+	 */
+	function getlocalesprocessor(codes) {
+		return (src, file) => {
+			const code = file.match(/ts-lang-(.*)\.js/)[1];
+			codes.add(code);
+			return (
+				`\tcase '${code}':\n` +
+				src
+					.split('\n')
+					.map(line => '\t\t' + line)
+					.join('\n') +
+				'break;'
+			);
+		};
+	}
+
+	/**
+	 * Moment.js has been copy-pasted from the internet with *all* the possible 
+	 * language code and that is considerably more than we need, so we'll remove 
+	 * the unneeded translations based on the list of supported language codes.
+	 * These translations fortunately follow a uniform pattern so that we can 
+	 * simply include or reject them based on advanced analysis of JS comments.
+	 * Note: The language code `nb` is equivalent to our wrongly name `no` code.
+	 * @param {Set<string>} codes
+	 * @returns {Function}
+	 */
+	function getmomentprocessor(codes) {
+		return src => {
+			const split = '//! moment.js locale configuration';
+			const short = [...codes].map(s => s.split('-')[0]);
+			const first = s => (s = s.trim()).slice(0, s.indexOf('\n'));
+			const trail = s => s.split('\n});')[1] || '';
+			const match = s => ((s = s.match(/\[(.*)\]/)) ? s[1] : null);
+			const patch = s => s === 'nb';
```

Comment:
shouldn't we just have a file called `nb` and another `no` so we support both?

Reply:
'no' doesn't exist, we made that up, that string is however emitted in V4 vis some obscure Grails config, so users will see the `nb` file if we create that; and also nobody will maintain it, or in best case maintain the wrong version 🙈

Reply:
What happens here is that we keep the `nb` locale in Moment and then we just-in-time translate `no` to `nb` during runtime.